src/jsx.d.ts

```diff
@@ -1480,9 +1480,13 @@ export namespace JSXInternal {
 	export type TargetedPictureInPictureEvent<Target extends EventTarget> =
 		TargetedEvent<Target, PictureInPictureEvent>;
 
+	export type EventHandlerObject<E extends TargetedEvent> = {
+		handleEvent(e: E): unknown
+	}
+
 	export type EventHandler<E extends TargetedEvent> = {
 		bivarianceHack(event: E): void;
-	}['bivarianceHack'];
+	}['bivarianceHack']  | EventHandlerObject<E>;
```

Comment:
This worked for my projects, but I'm still cautious of changing an exported type. It might be necessary to leave this intact and instead add to the attribute interfaces one-by-one.

Reply:
This seems to cause issues elsewhere (see #4581), though I'm not quite sure what it's even doing? If you have a moment, would love an explanation. A lot of this TS stuff flies right over my head

Reply:
I'll reply in the issue.

Reply:
Cheers, and no rush! If that's incorrect usage then fair enough too! I'm very much unfamiliar with what this is doing.